lightningd/options.c

```diff
@@ -860,6 +860,9 @@ static void handle_minimal_config_opts(struct lightningd *ld,
 			       &ld->config_dir,
 			       "Set working directory. All other files are relative to this");
 
+	if (!path_is_abs(ld->config_dir))
+		ld->config_dir = path_join(ld, path_cwd(tmpctx), ld->config_dir);
```

Comment:
Actually, path_join() does the Right Thing if the path is already absolute, so no need for if() here.

Reply:
Thanks